### PR TITLE
GAWB-2517: Reinstate no billing projects message

### PR DESCRIPTION
**Pull request by MatthewBemis**
- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [ ] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/cljs/main/broadfcui/components/buttons.cljs [hunk 1]
```diff
@@ -19,7 +19,9 @@
         (map? disabled?)
         (@modal-constructor
          (if (= (:type disabled?) :error) :error :message)
-         (assoc disabled? :on-dismiss on-dismiss))))
+         (assoc disabled? :on-dismiss on-dismiss))
+        :else
```
Comment (MatthewBemis) [inline src/cljs/main/broadfcui/components/buttons.cljs]: this else catches divs (and other things...)...

is there a way to test for a div specifically?

**Comment by MatthewBemis** [comment]
note to self: the existing links in here were actually broken during a documentation overhaul so I need to update them

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/WorkspaceListPage.scala [hunk 2]
```diff
@@ -21,12 +21,14 @@ class WorkspaceListPage(implicit webDriver: WebDriver) extends AuthenticatedPage
...
+  private val noBillingProjectsModal = testId("no-billing-projects-message")
   private def workspaceLink(ns: String, n: String) = Link(s"$ns-$n-workspace-link")
   private def restrictedWorkspaceLabel(ns: String, n: String) = Label(s"restricted-$ns-$n")
 
-  def clickCreateWorkspaceButton(): CreateWorkspaceModal = {
+  def clickCreateWorkspaceButton(expectDisabled: Boolean = false): Option[CreateWorkspaceModal] = {
+    if(expectDisabled) assert(createWorkspaceButton.isStateDisabled)
     createWorkspaceButton.doClick()
-    await ready new CreateWorkspaceModal
+    if(!expectDisabled) Option(await ready new CreateWorkspaceModal) else None
```
Comment (MattPutnam) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/WorkspaceListPage.scala:31]: This is great!

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/user/RegistrationSpec.scala [hunk 3]
```diff
@@ -59,6 +60,36 @@ class RegistrationSpec extends FreeSpec with BeforeAndAfter with Matchers with W
...
+      nonProfitStatus = true,
+      principalInvestigator = "Nobody",
+      city = "Cambridge",
+      state = "MA",
+      country = "USA")
+
+    await ready new DataLibraryPage()
+
+    val listPage = new WorkspaceListPage()
+    listPage.open
```
Comment (MattPutnam) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/user/RegistrationSpec.scala]: Combine this into:
```scala
val listPage = new WorkspaceListPage().open
```
And I think the `validateLocation()` below is redundant then.

File: src/cljs/main/broadfcui/components/buttons.cljs (hunk 1 above)
Comment (MattPutnam) [reply src/cljs/main/broadfcui/components/buttons.cljs]: It also catches `false`/`nil`. This function won't get called when `disabled?` is falsy, but it still feels wrong.

There's a `renderable?` function in components.cljs, which you could put in place of the `string?` check above.

**Review by davidangb — COMMENTED** [review]
Code looks fine; I didn't review at a detailed level.

In order, my preference for how to handle URLs to pages owned by comms:

1. Define long-lived static urls a la bitly.com or goo.gl and embed those in the UI code or config. Let comms manage where those static urls redirect to, outside of the firecloud codebase.
2. Define urls in config, so we could theoretically change the urls without releasing the entire ui codebase.
3. Hardcode urls in the ui code, like you've done here. If we need to take this approach, at least remove them from config so we don't have orphans.

**Comment by MatthewBemis** [comment]
@davidangb @MattPutnam back to you for re-review

config updates are here:
https://github.com/broadinstitute/firecloud-develop/pull/732
https://github.com/broadinstitute/firecloud-develop/pull/733
https://github.com/broadinstitute/firecloud-develop/pull/734

**Review by davidangb — APPROVED** [review]
:+1: thanks for the changes! I did not test.

**Comment by MattPutnam** [comment]
👍

**Review by MattPutnam — APPROVED** [review]

**Pull request merged**